docs/assumptions.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Compiler assumptions
+id: assumptions
+---
+
+<script defer src="/js/babel.min.js"></script>
+<script type="module" src="/js/components/mini-repl.js"></script>
+<script type="module" src="/js/components/assumption-repl.js"></script>
+
+<style>
+.assumption-input {
+  display: none;
+}
+</style>
+
+## `arrayLikeIsIterable`
```

Comment:
What do you think about adding a short explanation? https://github.com/babel/website/pull/2502#discussion_r615103087 - maybe at least that this is added in 7.13 and it replaces what used to be loose